### ISSUE TITLE
Improve the implementation of the v1beta1 CRD dependency downgrade logic

### ISSUE THREAD

**Issue by everettraven** [kubernetes-sigs/kubebuilder#2710]
### What do you want to happen?

The logic that performs the dependency downgrades to support v1beta1 CRDs needs to be cleaned up. 

The following PR implemented a way to remove all indirect dependencies and repopulate them with `go mod tidy` but did not remove the now obsolete replacement of indirect dependencies. 
PR: https://github.com/kubernetes-sigs/kubebuilder/pull/2679

Along with this, there should be a warning logged that informs users of the fact that v1beta1 CRDs are not supported in Kubernetes v1.22+ and that as a result Kubebuilder is downgrading certain dependencies to enable this functionality.

**Comment by camilamacedo86** [comment]
The v1beta1 is deprecated and we do not want to encourage people to use it.
Projects built with will not work from k8s 1.22 so do you think that is valid we spend effort on these improvements?

**Comment by everettraven** [comment]
I agree that we shouldn't focus on this since it is being deprecated and removed in the future. Closing.
